### PR TITLE
PoC updater

### PR DESCRIPTION
**Pull request by tisnik**
# Description

Updater having ability to compare old report with new one.

## Type of change

- New feature (non-breaking change which adds functionality)

## Testing steps

N/A

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: PoC-updater/updater.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	newReportAsString types.ClusterReport) (bool, error) {
+
+	// try to unmarshal the old report
+	oldReport, err := unmarshalReport(oldReportAsString)
+	if err != nil {
+		log.Error().Err(err).Msg("Unmarshal old report")
+		return false, err
+	}
+
+	// try to unarshal the newest report
```
Comment (epapbak) [inline PoC-updater/updater.go:121]: unmarshal

Comment (tisnik) [reply PoC-updater/updater.go]: Fixed by ongoing commit

File: PoC-updater/updater.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+		fmt.Println()
+	}
+}
+
+// Function issuesEqual compares two issues from reports
+func issuesEqual(issue1 types.ReportItem, issue2 types.ReportItem) bool {
+	if issue1.Type == issue2.Type &&
+		issue1.Module == issue2.Module &&
+		issue1.ErrorKey == issue2.ErrorKey &&
+		bytes.Compare(issue1.Details, issue2.Details) == 0 {
```
Comment (epapbak) [inline PoC-updater/updater.go:101]: could use bytes.Equal(issue1.Details, issue2.Details).
[Docs](https://golang.org/pkg/bytes/#Compare) say to prefer Equal over Compare for equality comparison

File: PoC-updater/updater.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+		return true
+	}
+	return false
+}
+
+// Function CompareReports compare old OCP report with the newest one and
+// returns boolean flag indiciation that new report contain new issues and thus
```
Comment (epapbak) [inline PoC-updater/updater.go:108]: indicating

File: PoC-updater/updater.go (hunk 2 above)
Comment (tisnik) [reply PoC-updater/updater.go:101]: ah it already exist? cool I was waiting for this function from Go 1.10 or so :)

Comment (tisnik) [reply PoC-updater/updater.go]: fixed :)

File: PoC-updater/updater.go (hunk 3 above)
Comment (tisnik) [reply PoC-updater/updater.go]: fixed, thank you

File: PoC-updater/updater.go [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+// user needs to be informed about them.
+func CompareReports(
```
Comment (epapbak) [inline PoC-updater/updater.go:110]: +1! Does the job nicely =)

**Review by epapbak — APPROVED** [review]
Left some minor comments

**Pull request merged**